### PR TITLE
Let's implement!

### PR DESCRIPTION
**Pull request by awinogradov**
**WIP**

FYI: @developit @aweary @ljharb

So, I tried to do this job and found some problems:
- Enzyme in depths `ReactShallowRenderer.prototype.render` checks components by native method `.isValidElement()`.  It is problem because we need to path our components for it.
- Preact renders numbers as a string, but React not
- Updating state by `.setState()` from the Wrapper doesn't affect the component render, but it returns valid state by `.state()`
- https://github.com/airbnb/enzyme/blob/master/packages/enzyme-test-suite is not ready for use with third-party adapters

Roadmap:
- [x] - Static Renderer
- [x] - Shallow Wrapper
- [ ] - Mount Wrapper

### CODE REVIEW

**Comment by ljharb** [comment]
Regarding `.isValidElement()`; this is something that each individual adapter should be able to decide.

If enzyme *itself* is currently checking `React.isValidElement()`, we need to move that to the adapter interface ASAP.

File: __tests__/mount-rendrer.js [hunk 1]
```diff
@@ -1,12 +0,0 @@
-/** @jsx h */
```
Comment (msholty-fd) [inline __tests__/mount-rendrer.js]: Careful of the filename! `mount-rendrer.js` instead of `mount-renderer.js`

File: package-lock.json [hunk 2]
```diff
@@ -0,0 +1,6656 @@
+{
```
Comment (ljharb) [inline package-lock.json]: i'd recommend gitignoring this and setting `package-lock=false` in `.npmrc`; only apps should have lockfiles

**Comment by kontrollanten** [comment]
Thanks for working on this. What's left? I don't have any deep experience of neither preact or React, but I may be able to put some hours in contributing to this

**Comment by aweary** [comment]
@awinogradov I've merged this and added you as a collaborator on the project. I don't have any free time for OSS right now, so feel free to work on this at your discretion.

**Comment by developit** [comment]
@kontrollanten hit me up on the [Preact Slack](https://preact-slack.now.sh) if you're looking for help.

**Pull request merged**